src/lib.rs

```diff
@@ -319,6 +341,88 @@ pub trait MapInsert<K>: Collection {
 	fn insert(&mut self, key: K, value: Self::Item) -> Self::Output;
 }
 
+/// Mutable map that supports the entry api
+pub trait EntryApi: Keyed {
+	type Occ<'a>: OccupiedEntry<'a, K = Self::Key, V = Self::Item>
+	where
+		Self: 'a;
+	type Vac<'a>: KeyVacantEntry<'a, K = Self::Key, V = Self::Item>
+	where
+		Self: 'a;
+
+	/// Gets the given key's corresponding entry in the map for in-place manipulation.
+	///
+	/// # Examples
+	///
+	/// ```
+	/// use std::collections::HashMap;
+	/// use std::ops::Index;
+	/// use cc_traits::{EntryApi, entry_api::*, Get};
+	///
+	/// fn make_map() -> impl EntryApi<Key=char,Item=u32> + Index<&'static char, Output=u32> + Get<&'static char> { HashMap::new() }
+	/// let mut letters = make_map();
+	///
+	/// for ch in "a short treatise on fungi".chars() {
+	///     let counter = letters.entry(ch).or_insert(0);
+	///     *counter += 1;
+	/// }
+	///
+	/// assert_eq!(letters[&'s'], 2);
+	/// assert_eq!(letters[&'t'], 3);
+	/// assert_eq!(letters[&'u'], 1);
+	/// assert!(!letters.contains_key(&'y'));
+	/// ```
+	fn entry(&mut self, key: Self::Key) -> Entry<Self::Occ<'_>, Self::Vac<'_>>;
+}
+
+
+pub trait EntryRefApi<Q: ToOwned<Owned = Self::Key> + ?Sized>: Keyed {
```

Comment:
Again for consistency, `EntryRef` would be better I think.
You can also move it to the `entry_api` module that you've made, the root is already overcrowded.